### PR TITLE
Using Semantic Kernel ITextCompletion As ITextGeneration

### PR DESCRIPTION
**Pull request by xbotter**
## Motivation and Context (Why the change? What's the scenario?)

This seems somewhat recursive, to use Semantic Kernel as the infrastructure for Kernel Memory. However, the interfaces of both are very similar. So, I added an adapter that can directly use SK's ITextCompletion to maximize the reuse of LLM adapted for SK.

### CODE REVIEW

**Comment by dluc** [comment]
could you upgrade the code to work with SK RC2 please?

**Comment by xbotter** [comment]
Updated

File: examples/105-dotnet-SemanticKernel-TextCompletion/Program.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+var embeddingDeployment = Env.Var("AOAI_DEPLOYMENT_EMBEDDING");
+
+var memory = new KernelMemoryBuilder()
+    .WithSemanticKernelTextGenerationService(new AzureOpenAIChatCompletionService(chatDeployment, "", endpoint, apiKey))
+    .WithCustomEmbeddingGeneration(new AzureOpenAITextEmbeddingGeneration(embeddingDeployment, "", endpoint, apiKey))
+    .Build<MemoryServerless>();
+
+// using document form 203-dotnet-using-core-nuget
+
+await memory.ImportDocumentAsync("sample-SK-Readme.pdf", documentId: "doc001");
```
Comment (dluc) [inline examples/105-dotnet-SemanticKernel-TextCompletion/Program.cs:19]: I'm getting `Microsoft.KernelMemory.KernelMemoryException: File not found: 'sample-SK-Readme.pdf'` here (this project doesn't see artifacts from example 203, why not include the PDF?)

**Review by dluc — CHANGES_REQUESTED** [review]
I've made a few changes, mostly moving the files from Core to Abstractions. There's an error in the new example, easy fix

**Comment by dluc** [comment]
putting on hold this, until https://github.com/microsoft/kernel-memory/pull/189 is merged. Some important changes to accommodate third-party/OSS models.

**Comment by xbotter** [comment]
The new changes have brought some configurable options, and I will adapt the interface as soon as possible.

**Comment by xbotter** [comment]
@dluc updated, please review again.

**Comment by dluc** [comment]
Looks like there are more breaking changes coming from RC4, let's wait until next week to reduce churn

**Review by dluc — APPROVED** [review]

**Pull request merged**